cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestController.java

```diff
@@ -48,19 +71,110 @@ void setUp() throws IOException {
     @AfterEach
     void tearDown() throws IOException {
         mockHttpServer.shutdown();
+        if (preferencesBackedCookieStore != null) {
+            preferencesBackedCookieStore.writeCookiesToPreferences();
+        }
     }
 
     ApiConnectionInfo buildConnectionInfo() {
-        String baseUrl = String.format("http://localhost:%s", mockHttpServer.getPort());
+        String baseUrl = getRadarBaseUrl();
         return new ApiConnectionInfoBuilder(baseUrl).build();
     }
 
-    final String readJsonFile(String jsonPath) throws IOException {
-        URL resource = getClass().getClassLoader().getResource(jsonPath);
+    static CookieJarFactory.CookieJarSupplier login() throws IOException {
+        Preferences node = Preferences.userRoot().node("test").node("cwms").node("http_client");
+        preferencesBackedCookieStore = new PreferencesBackedCookieStore(node);
+        CookieJarFactory.CookieJarSupplier cookieJarSupplier = CookieJarFactory.preferenceBackedCookieJar(preferencesBackedCookieStore);
+        MockHttpServer mockHttpServer = MockHttpServer.create();
+        if (cookieJarSupplier.isCookieExpired(getCwmsAAABaseUrl(mockHttpServer), "JSESSIONIDSSO")) {
+            mockHttpServer.enqueue(readJsonFile("cwms_aaa/cwms_aaa_banner_agreement.html"));
+            mockHttpServer.enqueue(readJsonFile("cwms_aaa/cwms_aaa_login.json"));
+            ApiConnectionInfo apiConnectionInfo = buildCwmsAAALoginConnectionInfo(cookieJarSupplier, mockHttpServer);
+            new CwmsLoginController().login(apiConnectionInfo);
+        }
+        mockHttpServer.shutdown();
+        return cookieJarSupplier;
+    }
+
+    ApiConnectionInfo buildConnectionInfo(CookieJarFactory.CookieJarSupplier cookieJar) throws IOException {
+        String baseUrl = getRadarBaseUrl();
+        AuthCookieCallback callback = new CwmsAuthCookieCallback(buildCwmsAAALoginConnectionInfo(cookieJar, mockHttpServer), cookieJar);
+        return new ApiConnectionInfoBuilder(baseUrl)
+            .withCookieJarSupplier(cookieJar)
+            .withCookieAuthenticator(callback)
+            .build();
+    }
+
+    String getRadarBaseUrl() {
+        if (USE_MOCK) {
+            return String.format("http://localhost:%s", mockHttpServer.getPort());
+        } else {
+            return "https://leary.rmanet.com:8443/cwms-data/";
```

Comment:
the leary hostname seems like something that should be the default value for a class constant pulled from a System property so that it wasn't too hard to -D a new value.